cpp/src/arrow/util/thread_pool.cc

```diff
@@ -183,6 +247,105 @@ void SerialExecutor::RunLoop() {
   }
   state_->current_thread = {};
 }
+#else   // ARROW_ENABLE_THREADING
+bool SerialExecutor::RunTasksOnAllExecutors(bool once_only) {
```

Comment:
Why do we need this?
Can we disable (or error) features that need this method?

Reply:
I think pretty much anything that does I/O assumes that the i/o calls and serialexecutor calls run in parallel. Without this stuff, serialexecutors can't run at the same time as I/O things which breaks quite a lot of things including loading parquet files. Executors need to be able to assume that other things will happen if they don't have any tasks ready to run or else bad things happen.